src/frontend/src/components/MapLegends.jsx

```diff
@@ -68,28 +68,33 @@ const MapLegends = ({ direction, spacing, iconBtnProps, defaultTheme, valueStatu
     //     );
     //   })}
     // </CoreModules.Stack>
-
-    <div className="fmtm-flex fmtm-flex-col fmtm-gap-4">
-      {MapDetails.map((data, index) => {
-        return (
-          <div key={index} className="fmtm-flex fmtm-items-center fmtm-gap-3">
-            <div className="fmtm-border-[1px] fmtm-border-gray-200">
-              {data.type !== 'locked' ? (
-                <CoreModules.IconButton
-                  style={{ backgroundColor: data.color, borderRadius: 0 }}
-                  {...iconBtnProps}
-                  color="primary"
-                  component="label"
-                  className="fmtm-w-10 fmtm-h-10"
-                ></CoreModules.IconButton>
-              ) : (
-                <AssetModules.LockIcon sx={{ fontSize: '40px' }} />
-              )}
+    <div>
+      <div className="fmtm-flex fmtm-gap-3 fmtm-border-b-[1px] fmtm-pb-2 fmtm-mb-4">
+        <AssetModules.LegendToggleIcon className=" fmtm-text-primaryRed" sx={{ fontSize: '35px' }} />
+        <p className="fmtm-text-2xl">Map Legend</p>
+      </div>
+      <div className="fmtm-flex fmtm-flex-col fmtm-gap-4">
+        {MapDetails.map((data, index) => {
```

Comment:
@NSUWAL123  Can you make a seperate component for this legend ? As right now its packed with too much html elements and rendering multiple elements.